src/Features/ERDDAP/Platform/Forecasts/Page/index.tsx

```diff
@@ -1,44 +1,120 @@
+  const timeSeries = platform.properties.readings.find((r) => standardNames.has(r.data_type.standard_name))
+
+  return <LoadForecastInfo {...props} {...{ platform, forecast_type, timeSeries }} />
 }
 
-function mapStateToProps({ erddap }: StoreState): ReduxProps {
-  return {
-    forecasts: erddap.forecasts.forecasts,
+interface LoadInfoProps extends Props {
+  timeSeries?: PlatformTimeSeries
+}
+
+/**
+ * Load all the forecasts, and load a matching time series if available.
+ */
+const LoadForecastInfo: React.FunctionComponent<LoadInfoProps> = ({ timeSeries, forecast_type, ...props }) => {
+  const { isLoading: isLoadingForecasts, data: forecastsInfo } = useForecasts()
+  const { data: dataset } = useDataset(timeSeries)
+
+  const forecastInfo = forecastsInfo?.find((f) => f.forecast_type.toLowerCase().replace(" ", "_") === forecast_type)
+
+  if (isLoadingForecasts) {
+    return <Alert>Loading forecast metadata</Alert>
+  }
+
+  if (forecastInfo) {
+    return <LoadForecast {...props} {...{ timeSeries, dataset, forecast_type, forecastInfo }} />
   }
+
+  return <Alert color="warning">Error loading forecast info</Alert>
 }
 
-// "northward_wind", "eastward_wind", "sea_surface_temperature", "wind_speed_of_gust",
-// , , "tendency_of_air_pressure",
-// "dew_point_temperature", , "air_pressure_at_sea_level",
-// , ,
-// "visibility_in_air", , "sea_water_salinity",
-// "sea_water_density", "sea_water_electrical_conductivity", "sea_water_temperature",
-// , , "barometric_pressure",
-// "sea_water_velocity", "direction_of_sea_water_velocity",
-// "volume_fraction_of_oxygen_in_sea_water", "oxygen_saturation",
-// "fractional_saturation_of_oxygen_in_sea_water"
+interface LoadForecastProps extends LoadInfoProps {
+  forecastInfo: ForecastSource
+  dataset?: DataTimeSeries
+}
+
+/**
+ * Load the forecast itself
+ */
+const LoadForecast: React.FunctionComponent<LoadForecastProps> = ({
+  forecastInfo,
+  platform,
+  unit_system,
+  timeSeries,
+  dataset,
+  forecast_type,
+}) => {
+  const point = platform.geometry as Point
+  const [lon, lat] = point.coordinates
+
+  const { isLoading, data } = useForecast(lat, lon, forecastInfo)
+
+  if (isLoading) {
+    return <Alert>Loading forecast</Alert>
+  }
+
+  if (data) {
+    const forecastDataset: DataTimeSeries = {
+      timeSeries: data,
+      name: forecastInfo.name,
+      unit: forecastInfo.units,
+    }
+
+    const datasets = [forecastDataset]
 
+    if (timeSeries && dataset) {
+      const aDayAgo = new Date()
+      aDayAgo.setDate(aDayAgo.getDate() - 1)
+
+      datasets.push({
+        ...dataset,
+        timeSeries: dataset.timeSeries.filter((r) => aDayAgo < r.time),
+        name: `${timeSeries.dataset}: ${timeSeries.data_type.long_name}`,
+      })
+    }
+
+    return (
+      <Row>
+        <Col>
+          <h4>{forecastInfo.forecast_type} Forecast</h4>
+          <ForecastChart data={datasets} unit_system={unit_system} type={forecast_type} />
```

Comment:
Codacy found an issue: [Identifier 'unit_system' is not in camel case.](https://app.codacy.com/manual/gmri/Neracoos-1-Buoy-App/pullRequest?prid=6166252)